src/psyclone/tests/psyGen_test.py

```diff
@@ -2223,21 +2296,35 @@ def test_find_write_arguments_no_more_nodes():
         in str(excinfo.value))
 
 
-def test_find_w_args_multiple_deps():
+def test_find_w_args_multiple_deps(monkeypatch, annexed):
     '''_find_write_arguments should return as many halo exchange
     dependencies as the vector size of the associated field. This test
     checks that this is the case and that the returned objects are
-    what is expected.
+    what is expected. We test with annexed dofs is True and False as
+    different numbers of halo exchanges are created.
+
     '''
 
+    config = Config.get()
+    dyn_config = config.api_conf("dynamo0.3")
+    monkeypatch.setattr(dyn_config, "_compute_annexed_dofs", annexed)
+
     _, invoke_info = parse(
         os.path.join(BASE_PATH, "8.3_multikernel_invokes_vector.f90"),
         distributed_memory=True, api="dynamo0.3")
     psy = PSyFactory("dynamo0.3",
                      distributed_memory=True).create(invoke_info)
     invoke = psy.invokes.invoke_list[0]
     schedule = invoke.schedule
-    loop = schedule.children[7]
+    # create halo exchanges between the two loops via redundant
+    # computation
+    if annexed:
+        index = 1
+    else:
+        index = 4
+    rc_trans = Dynamo0p3RedundantComputationTrans()
```

Comment:
Because the halo exchanges that we were relying on to do our tests no longer exist due to the gh_inc optimisation. We therefore redundantly compute to add halo exchanges back in.

Reply:
Thanks for explaining.